src/actions/register.js

```diff
@@ -0,0 +1,59 @@
+import API from "../config/api";
+
+const LoginUser = token => {
+  return async dispatch => {
+    try {
+      const response = await fetch(API.user.login, {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        credentials: "include",
+        body: JSON.stringify(token)
+      });
+      const status = response.status;
+      if (status >= 200 && status < 300) {
+        const data = await response.json();
+        return {
+          err: false,
+          data
+        };
+      } else {
+        const data = await response.json();
+        throw new Error(data.message);
+      }
```

Comment:
i think for this flow you can call `const data = await response.json();` once
what about something like 
```
const data = await.response.json();
const err = (status >= 200 && status < 300);
return { err, data };
```
you reduce the complexity here and then handle the response in the container itself